r/tests/testthat/test-dataset.R

```diff
@@ -303,11 +303,68 @@ test_that("Other text delimited dataset", {
       filter(integer > 6) %>%
       summarize(mean = mean(integer))
   )
+})
+
+test_that("readr parse options", {
+  arrow_opts <- names(formals(CsvParseOptions$create))
+  readr_opts <- names(formals(readr_to_csv_parse_options))
+
+  # Arrow and readr parse options must be mutually exclusive, or else the code
+  # in `csv_file_format_parse_options()` will error or behave incorrectly. A
+  # failure of this test indicates that these two sets of option names are not
+  # mutually exclusive.
+  expect_equal(
+    intersect(arrow_opts, readr_opts),
+    character(0)
+  )
 
-  # Now with readr option spelling (and omitting format = "text")
-  ds3 <- open_dataset(tsv_dir, partitioning = "part", delim = "\t")
+  # With not yet supported readr parse options
+  # (remove this after ARROW-8631)
+  if (!"na" %in% readr_opts) {
+    expect_error(
+      open_dataset(tsv_dir, partitioning = "part", delim = "\t", na = "\\N"),
+      "supported"
+    )
+  }
```

Comment:
Provided that you agree, here's a suggestion version you can just accept

```suggestion
  # With not yet supported readr parse options (ARROW-8631)
  expect_error(
    open_dataset(tsv_dir, partitioning = "part", delim = "\t", na = "\\N"),
    "supported"
  )
```